Symfony/CS/Fixer.php

```diff
@@ -311,6 +312,48 @@ public function fixFile(\SplFileInfo $file, array $fixers, $dryRun, $diff, FileC
         return $fixInfo;
     }
 
+    /**
+     * @param \SplFileInfo     $file
+     * @param Tokens           $tokens
+     * @param FixerInterface[] $fixers
+     *
+     * @return string[]
+     */
+    public function fixFileTokens(\SplFileInfo $file, Tokens $tokens, array $fixers)
```

Comment:
Because the fixer reads the file by itself using `file_get_contents` which cannot be stubbed.
I.e. there is no API on the Fixer that supports given code as string and the fixers you want to work on that string.
That is why the AbstractFixerTestBase doesn't use the Fixer itself, but only the implementing FixerInterface object that is being tested.